lib/foreman_maintain/repository_manager/el.rb

```diff
@@ -0,0 +1,88 @@
+module ForemanMaintain::RepositoryManager
+  class El
+    include ForemanMaintain::Concerns::OsFacts
+    include ForemanMaintain::Concerns::SystemHelpers
+
+    def disable_repos(repo_ids)
+      if el7?
+        execute!("yum-config-manager #{config_manager_options(repo_ids, 'disable')}")
+      else
+        execute!("dnf config-manager #{config_manager_options(repo_ids, 'set-disabled')}")
+      end
+    end
+
+    def rhsm_disable_repos(repo_ids)
+      if rhsm_available?
+        execute!(%(subscription-manager repos #{rhsm_options(repo_ids, 'disable')}))
+      else
+        logger.info("subscription-manager is not available.\
+                     Using #{pkg_manager} config manager instead.")
+        disable_repos(repo_ids)
+      end
+    end
+
+    def enable_repos(repo_ids)
+      if el7?
+        execute!("yum-config-manager #{config_manager_options(repo_ids, 'enable')}")
+      else
+        execute!("dnf config-manager #{config_manager_options(repo_ids, 'enable')}")
+      end
+    end
+
+    def rhsm_enable_repos(repo_ids)
+      if rhsm_available?
+        execute!(%(subscription-manager repos #{rhsm_options(repo_ids, 'enable')}))
+      else
+        logger.info("subscription-manager is not available.\
+                     Using #{pkg_manager} config manager instead.")
+        enable_repos(repo_ids)
+      end
+    end
+
+    def rhsm_options(repo_ids, option)
+      repo_ids.map { |r| "--#{option}=#{r}" }.join(' ')
+    end
+
+    def config_manager_options(repo_ids, option)
+      if repo_ids.is_a?(String)
+        if repo_ids == '*'
+          return format_shell_args("--#{option}" => repo_ids.to_s)
+        end
+
+        "--#{option}=#{repo_ids}"
+      elsif repo_ids.is_a?(Array)
+        "--#{option} #{repo_ids.join(',')}"
+      end
```

Comment:
why not

```ruby
repo_ids_string = if repo_ids.is_a?(Array)
                    repo_ids.join(',')
                  else
                    repo_ids
                  end
format_shell_args("--#{option}" => repo_ids_string)
```

While other repo ids shouldn't contain special shell chars, it doesn't hurt to call `foreman_shell_args` on these too, and makes the code more compact.